### PR TITLE
#183 - Changes to accommodate v1.5 of the SAM file specification

### PR DESCRIPTION
**Pull request by nenewell**
Changes to SamFileHeader.java, ValidateSamFileTest.java, and test files in SamFileHeaderMergerTest to accommodate v1.5 of the SAM file spec.

### CODE REVIEW

File: src/tests/java/htsjdk/samtools/ValidateSamFileTest.java [hunk 1]
```diff
@@ -437,7 +437,7 @@ public void testHeaderVersion(final String version, final boolean expectValid) t
                 {"1.0", true},
                 {"1.3", true},
                 {"1.4", true},
-                {"1.5", false},
+                {"1.5", true}
```
Comment (nh13) [inline src/tests/java/htsjdk/samtools/ValidateSamFileTest.java]: We probably want to add ```{"1.6", false}``` to the list to make sure we fail on version 1.6.

**Comment by nh13** [comment]
Thanks @nenewell.  I would suggest you squash the two commits and modify the commit so it only changes the two ```.java``` files, and not the ```.idea``` and ```*iml``` files.

**Comment by nenewell** [comment]
Thank you - I’ve updated the pull request.

**Comment by nenewell** [comment]
I see this has failed a Travis build. A recent build of a pull request of mine for hellbender failed due to a security issue because I am an external source - this was resolved by lbergelson (#478).

**Comment by lbergelson** [comment]
I don't think this is due to the same issue.  It looks like it might be a transient issue with downloading something from an FTP site.  Rerunning to see if it goes away.

**Comment by lbergelson** [comment]
Rerun solved the issue.

**Comment by nenewell** [comment]
Thank you.

**Comment by vdauwera** [comment]
Does anything more need to be done on this? @geoffjentry, could you opine?

**Comment by geoffjentry** [comment]
This might just need to be rebased on top of current htsjdk master but I'm getting test failures in picard when I use this htsjdk

**Comment by nenewell** [comment]
I will rebase.

**Comment by nenewell** [comment]
samRoundTrip test is failing. I'll fix once I've updated my system with ant. I don't understand why the build passed earlier test runs...

**Comment by nenewell** [comment]
The samRoundTrip test failure is fixed, and "VN:1.4" has been replaced with "VN1.5" in the headers of 21 other test files, even though these do not cause htsjdk test failures, in case they may cause test failures in the tool builds. But testLazyInstance failed in one of the four Travis runs - a "lazy" date object seems to have gotten the time wrong by 130 seconds...

**Comment by geoffjentry** [comment]
I'm still seeing test failures with picard when using this htsjdk

**Comment by nenewell** [comment]
@geoffjentry, could you please send a log file with the errors to nenewell@comcast.net? 
It would be great if there were a way for me to build the tools together with htsjdk, so you folks wouldn't have to be bothered with this.

**Comment by geoffjentry** [comment]
@nenewell You should be able to do it, I'm talking about the publicly available picard tools (https://github.com/broadinstitute/picard).

If you clone that repo and then have your htsjdk sitting in that top level dir you should be able to run 'ant test'

**Comment by nenewell** [comment]
Thanks - I will do that.

**Comment by geoffjentry** [comment]
@nenewell Great. Let me know if you encounter difficulties

**Comment by nenewell** [comment]
The picard test failures were caused by the presence of the old version number in the VN fields of test files - I've updated those and submitted the changes as picard pull request #211. So picard should be OK - I still get six test failures, but they must be related to my setup and not the SAM file version update because I also get them when I use unmodified htsjdk and picard code from the Broad trunk. I can't access non-picard tool development, but any version-specific test files there will also need to be updated...

**Comment by nenewell** [comment]
The current state on this, to the best of my understanding, is that the necessary htsjdk updates are handled by pull request 233, and the necessary picard updates are handled by picard pull request 211 (which fails Travis because htsjdk changes are not yet merged). I don't know if any other tools need updating - if so, I'd be happy to do that, but I don't know if I would be able to access them. Please let me know if you'd like me to rebase this pull request and picard request 211, or if there's anything else I can do on this...Thank you

**Comment by jmarshall** [comment]
Is the change to `ACCEPTABLE_VERSIONS` at least likely to be merged any time soon?  Fortunately samtools does not yet generate VN:1.5 either, but it's been in the spec for two years and probably other tools do…

**Comment by nh13** [comment]
@nenewell looks like #211 includes a patch for hdfs, which goes beyond just fixing tests for the new version in picard.

**Comment by nenewell** [comment]
The link in my post of May 19 points to htsjdk #211 (not picard 211) due to git’s default behavior –  I didn’t yet know that prepending ‘#’ to a number automatically produces a link to that request in the current repo.

**Comment by nh13** [comment]
Agreed it was my mistake as I deleted the comment after I made it. 

Thumb typed for added typos

**Comment by akiezun** [comment]
@geoffjentry can you review and decide what @nenewell needs to do here?

**Comment by nh13** [comment]
Note: Impact to production systems need to be considered and tested before this is rolled out.  How will projects that have mixes of 1.4 versus 1.5 work etc?  I am sure others may face the same problem.  This is an internal Broad question to answer, so lets take it offline, but it unit tests to verify our desired behavior in the case seem necessary.

**Comment by nh13** [comment]
Closing this and moving the PR discussion to #285

**Pull request closed without merging**